src/SegmentationCameraSensor.cc

```diff
@@ -426,7 +440,7 @@ bool SegmentationCameraSensor::Update(
   // Save image
   if (this->dataPtr->saveImage)
   {
-    this->dataPtr->SaveImage(
+    this->dataPtr->SaveImage(this->dataPtr->segmentationLabelsBuffer,
       this->dataPtr->segmentationColoredBuffer, width, height);
```

Comment:
```suggestion
    this->dataPtr->SaveImage(this->dataPtr->segmentationColoredBuffer,
      this->dataPtr->segmentationLabelsBuffer, width, height);
```
Looking at the `SaveImage` method, the colored buffer is the first parameter, not the labels buffer https://github.com/ignitionrobotics/ign-sensors/blob/4ac74b351c91fa0dfde1ad93b819b07590159ed2/src/SegmentationCameraSensor.cc#L38-L48